tools/run-test-perf.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+trap "exit" INT
+
+function pr_err() {
+	echo -e "\e[91mError: $@\e[39m"
+}
+
+function exit_err() {
+	pr_err $@
+	exit
+}
+
+USAGE="Usage:\n    sudo bash run.sh OLD_ENGINE NEW_ENGINE REPEATS"
+
+if [ "$(whoami)" != "root" ]
+then
+	pr_err "start script with sudo..."
+	echo -e $USAGE
+	exit
+fi
+
+if [ "$#" -ne 3 ]
+then
+	pr_err "Argument number mismatch..."
+	echo -e $USAGE
+	exit
+fi
+
+ENGINE_OLD="$1"
+ENGINE_NEW="$2"
+REPEATS=$3
+
+if [ $REPEATS -lt 1 ]
+then
+	exit_err "REPEATS must be greater than 0"
+fi
+
+if [ ! -x $ENGINE_OLD ]
+then
+	exit_err "\"$ENGINE_OLD\" is not an executable file..."
+fi
+
+if [ ! -x $ENGINE_NEW ]
+then
+	exit_err "\"$ENGINE_NEW\" is not an executable file..."
+fi
+
+#check if all required programs are exist
+REQUIRED_PROGS="awk bash bc cpufreq-set printf renice taskset time timeout "
+MISSING_PROGS=""
+
+for program in $REQUIRED_PROGS
+do
+	if [ -z $(command -v $program) ]
+	then
+		MISSING_PROGS="$MISSING_PROGS $program"
+	fi
+done
+
+if [ -n "$MISSING_PROGS" ]
+then
+	pr_err "install missing programs and try again:"
+	for program in $MISSING_PROGS
+	do
+		echo "   $program"
+	done
+	exit
+fi
+
+SINGLE_TEST_TIMEOUT="65"
+TIMEOUT=$(echo "$3" "${SINGLE_TEST_TIMEOUT}" | awk '{print $1 * $2;}')
+CPU=3
+
+RAND_VA_SPACE="/proc/sys/kernel/randomize_va_space"
+echo 0 > $RAND_VA_SPACE || exit_err "setting $RAND_VA_SPACE"
+
+renice -n -19 $$ || exit_err "renice unsuccedeed"
+#cpufreq-set -g performance || exit_err "couldn't set performance governor"
+cpufreq-set -c ${CPU} -f 500MHz
```

Comment:
This is mostly usable for arm board like Arndale or Rasp PI: Not to overhead them and force predicable frequency between different runs.